services/repository/files/content.go

```diff
@@ -219,7 +219,7 @@ func GetContents(ctx context.Context, repo *repo_model.Repository, treePath, ref
 		}
 	}
 	// Handle links
-	if entry.IsRegular() || entry.IsLink() {
+	if entry.IsRegular() || entry.IsLink() || entry.IsExecutable() {
```

Comment:
How are directories stored in git?
As 755?
If yes, then is it reported as `executable` too?

Reply:
Since there is no response from author, I would like to try to answer it .... I think it is right, because the entry mode is determined by separate constants (not by bit masks):

![image](https://github.com/go-gitea/gitea/assets/2114189/bdf7f7b7-e2ed-46e2-978f-5890ada02df5)